pkg/devspace/deploy/deployer/kubectl/builder.go

```diff
@@ -81,7 +83,30 @@ func (k *kubectlBuilder) Build(manifest string, cmd RunCommand) ([]*unstructured
 		args = append(args, "--namespace", k.namespace)
 	}
 
-	args = append(args, "--dry-run", "--output", "yaml", "--validate=false")
+	// compare kubectl version for --dry-run flag value
+	out, err := command.NewStreamCommand(k.path, []string{"version", "--client", "--short"}).Output()
+	if err != nil {
+		return nil, err
+	}
+
+	versionSlice := strings.Split(string(out), "v")
+	v1, err := version.NewVersion(strings.TrimSpace(versionSlice[1]))
```

Comment:
use `strings.TrimPrefix(string(out), "v")` here instead of split & trimSpace

Reply:
This way is not working. Not sure why but it is not returning a trimmed value.

Reply:
And there're trailing space in out so anyhow we'll need to use trimSpaces.

Reply:
Then use `version.NewVersion(strings.TrimPrefix(strings.TrimSpace(string(out)), "Client Version: v"))`, the problem with the split approach is that you expect that there will be a v..., but there might be situations where this is not the case, for example if you self-compile kubectl or if they change the way the version is printed, we shouldn't panic here